src/appleseed.studio/mainwindow/mainwindow.cpp

```diff
@@ -300,6 +300,7 @@ bool MainWindow::pack_project(QString filepath)
 
 void MainWindow::close_project()
 {
+    remove_render_tabs();
```

Comment:
This is done to remove the reference to `Lazy<renderer::TriangleTree>` before it gets destroyed (as the project is getting destroyed by the next line `m_project_manager.close_project()`) so that [this](https://github.com/appleseedhq/appleseed/blob/master/src/appleseed/foundation/utility/lazy.h#L369) `assert` don't go off.

_Why is there a reference to `Lazy<renderer::TriangleTree>` ?_
`RenderTab` has a `ScenePicker` member.
`ScenePicker` has an `Intersector` member.
`Intersector` has a `TriangleTreeAccessCache` member.
`TriangleTreeAccessCache` is defined as,
```
typedef foundation::AccessCacheMap<
    TriangleTreeContainer,
    TriangleTreeAccessCacheLines,
    TriangleTreeAccessCacheWays,
    foundation::PoolAllocator<void, TriangleTreeAccessCacheLines * TriangleTreeAccessCacheWays>
> TriangleTreeAccessCache;
```
Note that it has a `TriangleTreeContainer`, which is defined as,
```
typedef std::map<
    foundation::UniqueID,
    foundation::Lazy<TriangleTree>*
> TriangleTreeContainer;
```
Note that it has `foundation::Lazy<TriangleTree>*` which doesn't let the `foundation::Lazy<TriangleTree>` to get destroyed, because the aforementioned `assert` goes off.

For more clarification, I'm including the call stack of what happens if I __remove__ the call to `remove_render_tabs`

![callstack](https://user-images.githubusercontent.com/35342732/77248456-76db5500-6c5f-11ea-9165-896f61c83526.PNG)

I hope it makes things clearer. As usual, if I'm doing something wrong please let me know. :)

Reply:
Wow, great explanations !

Reply:
Is this change still relevant?

Reply:
Yes, it'll be relevant as long as we have `ScenePicker` as a member of `RenderTab`.